src/Microsoft.DotNet.ProjectModel/Resolution/PackageDependencyProvider.cs

```diff
@@ -40,8 +45,8 @@ public PackageDescription GetDescription(NuGetFramework targetFramework, LockFil
             var dependencies = new List<LibraryRange>(targetLibrary.Dependencies.Count + targetLibrary.FrameworkAssemblies.Count);
             PopulateDependencies(dependencies, targetLibrary, targetFramework);
 
-            var path = _packagePathResolver.GetInstallPath(package.Name, package.Version);
-            var exists = Directory.Exists(path);
+            var path = _packagePathResolver?.GetPackageDirectory(package.Name, package.Version);
+            bool exists = path != null && Directory.Exists(path);
```

Comment:
packagePathResolver checks that the hash file exists, so the directory check is already done. You can just check for null here.